src/Machine.Specifications/Catch.cs

```diff
@@ -1,30 +1,61 @@
 using System;
+using System.Threading.Tasks;
 
 namespace Machine.Specifications
 {
     public static class Catch
     {
         public static Exception Exception(Action throwingAction)
         {
-            return Only<Exception>(throwingAction);
+            try
+            {
+                throwingAction();
+            }
+            catch (Exception ex)
+            {
+                return ex;
+            }
+
+            return null;
         }
 
         public static Exception Exception<T>(Func<T> throwingFunc)
         {
+            Task task;
+
             try
             {
-                throwingFunc();
+                task = throwingFunc() as Task;
             }
             catch (Exception exception)
             {
                 return exception;
             }
 
+            if (task != null)
+            {
+                throw new InvalidOperationException("You must use Catch.ExceptionAsync for async methods");
+            }
+
+            return null;
+        }
+
+        public static async Task<Exception> ExceptionAsync(Func<Task> throwingAction)
+        {
+            try
+            {
+                await throwingAction();
+            }
+            catch (Exception ex)
+            {
+                return ex;
+            }
+
             return null;
         }
 
         public static TException Only<TException>(Action throwingAction)
-          where TException : Exception
+            where TException : Exception
```

Comment:
Should Only also be OnlyAsync ?

Reply:
No, I'm going to be deprecating the `Only` method and removing it entirely in a later major version. It breaks the AAA pattern. Will write a longer post when I make the change. But see here for starters: https://github.com/xunit/xunit/issues/188#issuecomment-104378832